plugins/aea-ledger-hwi/aea_ledger_hwi/account.py

```diff
@@ -0,0 +1,356 @@
+
+class HWIAccount:
+    """Hardware wallet interface as ethereum account"""
+
+    default_account: Optional[HWIAccountData]
+
+    def __init__(
+        self,
+        default_device: int = 0,
+        default_key_index: int = 0,
+    ) -> None:
+        """Initialize object."""
+
+        self.default_device = default_device
+        self.default_key_index = default_key_index
+        self.default_account = None
+
+    @property
+    def devices(self) -> List[Device]:
+        """Returns the list of available devices."""
+
+        return enumerate_devices()
+
+    def get_client(
+        self,
+        device_index: Optional[int] = None,
+    ) -> LedgerClient:
+        """Get ledger client."""
+        device_index = device_index or self.default_device
+        return LedgerClient(device=self.devices[device_index])
+
+    def get_account(
+        self,
+        key_index: Optional[int] = None,
+        device_index: Optional[int] = None,
+    ) -> HWIAccountData:
+        """Get hardware wallet account."""
+
+        key_index = key_index or self.default_key_index
+        path = m / h(44) / h(60) / h(key_index) / 0 / 0
+        opts = GetEthPublicAddressOpts(display_address=True, return_chain_code=True)
+
+        path_construct = PrefixedArray(Byte, Int32ub)
+        path_apdu = path_construct.build(path.to_list())
+        client = self.get_client(device_index=device_index)
+        response = client.apdu_exchange(
+            ins=0x02,
+            data=path_apdu,
+            p1=0x01 if opts.display_address else 0x00,
+            p2=0x01 if opts.return_chain_code else 0x00,
+        )
+
+        struct_kwargs = dict(
+            public_key=Prefixed(Int8ub, GreedyBytes),
+            address=PascalString(Int8ub, "ascii"),
+        )
+        if opts.return_chain_code:
+            struct_kwargs["chain_code"] = Bytes(32)
+
+        response_template = Struct(**struct_kwargs)
+
+        parsed_response = response_template.parse(response)
+        return HWIAccountData(
+            public_key=parsed_response.public_key,
+            address=("0x" + parsed_response.address),
+            chain_code=parsed_response.chain_code if opts.return_chain_code else None,
+        )
+
+    @property
+    def address(self) -> ChecksumAddress:
+        """Address"""
+        if self.default_account is None:
+            self.default_account = self.get_account()
+        return self.default_account.address
+
+    @property
+    def public_key(self) -> ChecksumAddress:
+        """Public key"""
+        if self.default_account is None:
+            self.default_account = self.get_account()
+        return self.default_account.public_key
+
+    def sign_message(
+        self, signable_message: SignableMessage, **kwargs: Any
+    ) -> SignedMessage:
+        """Sign a EIP191 message"""
+
+        raise NotImplementedError()
+
+    @staticmethod
+    def encode_transaction(
+        transaction: TypedTransaction,
+        is_eip1559_tx: bool = False,
```

Comment:
why not make eip1559 the default?

Reply:
https://github.com/valory-xyz/open-aea/blob/c7458fe2581f96eed5e8b1355295056d620a6ff2/plugins/aea-ledger-hwi/aea_ledger_hwi/account.py#L253-L262

Currently this method is being used internally and the type of transaction is defined automatically based on what parameters are provided.